### DISCUSSION TITLE
ZKsync contract deployment error

### DISCUSSION THREAD

**Discussion by DevinBeu** [Cyfrin/foundry-full-course-cu#2903] [General]
While using remix i added the Zksync extension. Added the simpleStorage.sol file and compiled it. I am having this error when trying to deploy. I made sure to recieve the zksync testnet funds and have my metamask connected. 
![zksyncProb](https://github.com/user-attachments/assets/a1364317-1978-4d74-aa91-bc325ef0ae1c)

**Comment by EngrPips** [comment]
You need to connect Remix to your metamask wallet so as to deploy using the account that has the `zksync` sepolia tokens. The images below should guide you on how to connect your Remix to Metamask, but if you encounter any issues, please ask questions.

![Screenshot 2024-11-22 at 08 03 13](https://github.com/user-attachments/assets/d6143972-da27-4d71-8b7b-40198e10b09a)


![Screenshot 2024-11-22 at 08 03 26](https://github.com/user-attachments/assets/34749466-1af6-471b-9092-26910bf23abc)


![Screenshot 2024-11-22 at 08 03 46](https://github.com/user-attachments/assets/9f23f35a-aab8-4b71-a907-feae7a97bc59)


![Screenshot 2024-11-22 at 08 04 01](https://github.com/user-attachments/assets/fffc24c8-89a1-4da6-afc9-bf9867b9f3e8)


![Screenshot 2024-11-22 at 08 04 11](https://github.com/user-attachments/assets/ae98b40d-2b94-4ed1-836a-c9cce04d7be9)

**Reply by DevinBeu** [reply]
thanks!

**Reply by EngrPips** [reply]
Anytime.
